File: csharp/DVSClient/DVSClient/address/format/components/SubBuilding.cs
using DVSClient.Server.Address.Format;

namespace DVSClient.Address.Format.Components
{
    public class SubBuilding
    {
        public string Name { get; }
        public SubBuildingItem? Entrance { get; }
        public SubBuildingItem? Floor { get; }
        public SubBuildingItem? Door { get; }

        public SubBuilding(RestApiAddressFormatComponents.RestApiAddressComponentSubBuilding api)
        {
            Name = api.Name ?? string.Empty;
            Entrance = api.Entrance != null ? new SubBuildingItem(api.Entrance.FullName, api.Entrance.Type, api.Entrance.Value) : null;
            Floor = api.Floor != null ? new SubBuildingItem(api.Floor.FullName, api.Floor.Type, api.Floor.Value) : null;
            Door = api.Door != null ? new SubBuildingItem(api.Door.FullName, api.Door.Type, api.Door.Value) : null;
        }
    }
}

File: java/src/main/java/com/experian/dvs/client/address/format/components/SubBuilding.java
package com.experian.dvs.client.address.format.components;

import com.experian.dvs.client.server.address.format.RestApiAddressFormatComponents;

import java.util.Objects;

public class SubBuilding {
    private final String name;
    private final SubBuildingItem entrance;
    private final SubBuildingItem floor;
    private final SubBuildingItem door;

    // Constructor
    public SubBuilding(RestApiAddressFormatComponents.RestApiAddressComponentSubBuilding api) {
        this.name = Objects.toString(api.getName(), "");
        this.entrance = api.getEntrance() != null ? new SubBuildingItem(api.getEntrance().getFullName(), api.getEntrance().getType(), api.getEntrance().getValue()) : null;
        this.floor = api.getFloor() != null ? new SubBuildingItem(api.getFloor().getFullName(), api.getFloor().getType(), api.getFloor().getValue()) : null;
        this.door = api.getDoor() != null ? new SubBuildingItem(api.getDoor().getFullName(), api.getDoor().getType(), api.getDoor().getValue()) : null;
    }

    // Getters
    public String getName() {
        return name;
    }

    public SubBuildingItem getEntrance() {
        return entrance;
    }

    public SubBuildingItem getFloor() {
        return floor;
    }

    public SubBuildingItem getDoor() {
        return door;
    }
}
